src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -65,6 +74,53 @@ public List<ExtensionRestHandler> getExtensionRestHandlers() {
             );
     }
 
+    @Override
+    public Collection<Object> createComponents(ExtensionsRunner runner) {
+
+        SDKRestClient sdkRestClient = getRestClient();
+        SDKClusterService sdkClusterService = runner.getSdkClusterService();
+        Settings environmentSettings = runner.getEnvironmentSettings();
+        NamedXContentRegistry xContentRegistry = runner.getNamedXContentRegistry().getRegistry();
```

Comment:
Make this `SDKNamedXContentRegistry` and pass that to the constructors of the objects.  Don't do the final `getRegistry()` call here.  Call `getRegistry()` internally in the objects you are instantiating when you need a `NamedXContentRegistry` object.